File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/template/ConcreteSub.java
package designPatterns.behavioral.template;

public class ConcreteSub {
  static class A extends Super {
    @Override
    String initialize() {
      return "A initialized!";
    }

    @Override
    String startPlay() {
      return "A started.";
    }

    @Override
    String endPlay() {
      return "A finished!";
    }
  }

  static class B extends Super {
    @Override
    String initialize() {
      return "B initialized!";
    }

    @Override
    String startPlay() {
      return "B started.";
    }

    @Override
    String endPlay() {
      return "B finished!";
    }
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/template/ConcreteSub.kt
package behavioral.template

sealed class ConcreteSub {
  class A : Super() {
    override fun initialize(): String = "A initialized!"
    override fun startPlay(): String = "A started."
    override fun endPlay(): String = "A finished!"
  }

  class B : Super() {
    override fun initialize(): String = "B initialized!"
    override fun startPlay(): String = "B started."
    override fun endPlay(): String = "B finished!"
  }
}
